### PR TITLE
Implement very simple first version of create batch spec execution page

### PR DESCRIPTION
**Pull request by eseliger**
So far this is just a simpler entry point to passing a spec file to the backend.

image

### CODE REVIEW

**Review by malomarrec — APPROVED** [review]
🚀

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 1]
```diff
@@ -137,9 +160,39 @@ export const CreateBatchChangePage: React.FunctionComponent<CreateBatchChangePag
...
+                            className="btn btn-primary"
+                            onClick={submitBatchSpec}
+                            disabled={isLoading === true}
+                        >
+                            Run batch spec
+                        </button>
+                        {batchSpecExecution && (
+                            <div className="mt-3 mb-0 alert alert-success">
+                                Running batch spec.{' '}
+                                <Link to={`/users/erik/batch-changes/executions/${batchSpecExecution.id}`}>
```
Comment (LawnGnome) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:188]: This link probably shouldn't have a hardcoded user.

File: schema/batch_spec.schema.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "$id": "batch_spec.schema.json#",
```
Comment (LawnGnome) [inline schema/batch_spec.schema.json:2]: I believe this needs to be an absolute URL. (Or, at least, it's strongly recommended that it is.)

**Review by LawnGnome — APPROVED** [review]

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:188]: Yeah. I don't have the namespace and the routes set up yet, so the link is fake. I'll do that in the next PR!

File: schema/batch_spec.schema.json (hunk 2 above)
Comment (eseliger) [reply schema/batch_spec.schema.json:2]: Hmm. This is what we always do for our json schemas, it is unique, at least. Not sure what the impact is when it _isn't_ a URL.

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 3]
```diff
@@ -2,15 +2,22 @@ import classNames from 'classnames'
...
 import { Container, PageHeader } from '@sourcegraph/wildcard'
 
+import batchSpecSchemaJSON from '../../../../../../schema/batch_spec.schema.json'
 import { BatchChangesIcon } from '../../../batches/icons'
+import { ErrorAlert } from '../../../components/alerts'
 import { PageTitle } from '../../../components/PageTitle'
 import { SidebarGroup, SidebarGroupHeader, SidebarGroupItems } from '../../../components/Sidebar'
+import { BatchSpecExecutionFields } from '../../../graphql-operations'
 import { Settings } from '../../../schema/settings.schema'
+import { MonacoSettingsEditor } from '../../../settings/MonacoSettingsEditor'
```
Comment (courier-new) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:18]: Nice!

File: schema/batch_spec.schema.json (hunk 2 above)
Comment (courier-new) [reply schema/batch_spec.schema.json:2]: Total n00b question here, but what exactly is this change for? I found the [schema documentation](https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.8.2.1) on the `$id` keyword, but I think I'm missing why this wasn't required before but is for some reason needed now?

Comment (eseliger) [reply schema/batch_spec.schema.json:2]: monaco requires the `$id` IIRC. It is also part of the types for the monaco settings editor

Comment (courier-new) [reply schema/batch_spec.schema.json:2]: Ah okay, thanks!

**Pull request merged**